lib/helper/Appium.js

```diff
@@ -0,0 +1,839 @@
+  }
+
+  /**
+   * Check the device orientation
+   * Appium: support Android and iOS
+   */
+  checkOrientationIs(orientation) {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.orientation().then(function(res) {
+        return truth('orientation', `to be ${orientation}`).assert(res.value == orientation);
+      });
+    } else
+      throw new Error(`getDeviceOrientation method can be used only with apps`);
+  }
+
+  /**
+   * Close the given application.
+   * Appium: support only iOS
+   */
+  closeApp() {
+    if (this.options.desiredCapabilities.platformName == "iOS") {
+      return this.browser.closeApp();
+    } else
+      throw new Error(`closeApp method can be used only with iOS apps`);
+  }
+
+  /**
+   * Get list of all available contexts
+   * Appium: support Android and iOS
+   */
+  grabAllContexts() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.contexts();
+    } else
+      throw new Error(`getAllContexts method can be used only with apps`);
+  }
+
+  /**
+   * Retrieve current context
+   * Appium: support Android and iOS
+   */
+  grabContext() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.context();
+    } else
+      throw new Error(`getContext method can be used only with apps`);
+  }
+
+  /**
+   * Get current device activity.
+   * Appium: support only Android
+   */
+  grabCurrentActivity() {
+    if (this.options.desiredCapabilities.platformName == "Android") {
+      return this.browser.getCurrentDeviceActivity();
+    } else
+      throw new Error(`getCurrentDeviceActivity method can be used only with Android apps`);
+  }
+
+  /**
+   * Get the current geolocation.
+   * Only for browser, firstly you sould setGeoLocation
+   * Appium: support Android and iOS
+   */
+  grabGeoLocation() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.getGeoLocation();
+    } else
+      throw new Error(`getGeoLocation method can be used only with apps`);
+  }
+
+  /**
+   * Get informations about the current network connection (Data/WIFI/Airplane).
+   * The actual server value will be a number. However WebdriverIO additional
+   * properties to the response object to allow easier assertions.
+   * Appium: support only Android
+   */
+  grabNetworkConnection() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.getNetworkConnection();
+    } else
+      throw new Error(`getNetworkConnection method can be used only with Android apps`);
+  }
+
+  /**
+   * Get current orientation.
+   * Appium: support Android and iOS
+   */
+  grabOrientation() {
+    if (this.options.desiredCapabilities.platformName) {
+      return this.browser.orientation().then(function(res) {
+        return res.value;
+      });
+    } else
+      throw new Error(`grabOrientation method can be used only with apps`);
```

Comment:
There are lots of typical errors here, we can move them out into special method and reusue them.

```js
// in the end of a file
function onlyForApps(name) {
   if (!this.options.desiredCapabilities.platformName) {
      throw new Error(`${name} method can be used only with apps`)
  }
}
```

inside `grabOrientation` and others:

```js
grabOrientation() {
   onlyForApps.call(this, arguments.calee.name);
   // continue...
```